cookbooks/fb_networkmanager/recipes/default.rb

```diff
@@ -0,0 +1,100 @@
+#
+# Cookbook:: fb_networkmanager
+# Recipe:: default
+#
+# Copyright (c) 2019-present, Vicarious, Inc.
+# Copyright (c) 2019-present, Facebook, Inc.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+vpnhome = '/var/lib/openvpn/chroot'
+FB::Users.initialize_group(node, 'nm-openvpn')
+node.default['fb_users']['users']['nm-openvpn'] = {
+  'gid' => 'nm-openvpn',
+  # If /var/lib/openvpn doesn't exist yet, we can't create it in time
+  # but we want to create the user, so for the first run, create it
+  # with /tmp
+  'home' => ::File.directory?(vpnhome) ? vpnhome : '/tmp',
+  'shell' => '/usr/sbin/nologin',
+  'action' => :add,
+}
+
+packages = %w{
+  network-manager
+  openvpn
+  network-manager-openvpn-gnome
+}
+
+package packages do
+  only_if { node['fb_networkmanager']['enable'] }
+  action :upgrade
+end
+
+template '/etc/NetworkManager/NetworkManager.conf' do
+  only_if { node['fb_networkmanager']['enable'] }
+  owner 'root'
+  group 'root'
+  mode '0644'
+  source 'nm.conf.erb'
+  helper(:data) { node['fb_networkmanager']['config'] }
```

Comment:
TIL. Is this a Chef 15 thing?

Reply:
Nope, I think it's been around since 13. I don't love using it, because I'd rather people use the node directly, but since I'm pulling from very different places in the node it seemed a bit cleaner. If this is undesired, instead I can pass in something like:

```ruby
variables(:type => :global)
```

here and:

```ruby
variables(:type => :system, :name => connection)
```

below and then just grab the right part of the node inside of the tempalte.